### PR TITLE
API middleware implementation

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related Issues
closes https://github.com/DivanteLtd/vue-storefront/issues/4854

### CODE REVIEW

File: packages/core/core/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   "dependencies": {
     "@vue/composition-api": "1.0.0-beta.14",
     "vue": "^2.6.11",
-    "lodash-es": "^4.17.15"
+    "lodash-es": "^4.17.15",
+    "express": "^4.17.1"
```
[the remark this answers is not in the document]
Comment (andrzejewsky) [reply packages/core/core/package.json:20]: it's even bundled separately, but I also checked - no express at all

File: packages/core/core/rollup-server.config.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import pkg from './package.json';
+import typescript from 'rollup-plugin-typescript2';
+import replace from '@rollup/plugin-replace';
+
+export function generateBaseConfig(pkg) {
+  return {
+    input: 'src/server/index.ts',
+    output: [
+      {
+        file: pkg.mainServer,
```
Comment (filrak) [inline packages/core/core/rollup-server.config.js:10]: for the server we just need this one, no benefit in having es build here

File: packages/core/docs/integrate/api-middleware.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# API middleware
```
Comment (filrak) [inline packages/core/docs/integrate/api-middleware.md:1]: I menath - shouldn't that be a part of `/integrate/ecommerce' doc?

File: packages/core/docs/integrate/api-middleware.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+The function `createMiddleware` returns two properties:
+- `middleware` - which is the created middleware used directly by `addServerMiddleware`
+- `extend` - a function that adds a new endpoint to the middleware. It takes as an argument an express.js instance
+
+## Nuxt module configuration
+
+Once we have created our middleware in the integration, `createMiddleware` exposes also config option, called `extendApi`. It works in the same way as `extend` - gives us a possibility to extend api anywhere else in the project.
+
+```js
+['@vue-storefront/commercetools/nuxt', {
```
Comment (filrak) [inline packages/core/docs/integrate/api-middleware.md:35]:
```suggestion
['@vue-storefront/{PLATFORM}/nuxt', {
```

File: packages/core/docs/.vuepress/config.js [hunk 5]
```diff
@@ -116,7 +116,9 @@ module.exports = {
             ['/general/architecture', 'Architecture'],
             ['/general/i18n', 'i18n'],
             ['/general/faceting', 'Faceting'],
-            ['/general/logging', 'Logging']
+            ['/general/logging', 'Logging'],
+            ['/general/api-middleware', 'API middleware']
```
Comment (filrak) [inline packages/core/docs/.vuepress/config.js:120]:
```suggestion
```

File: packages/core/docs/.vuepress/config.js [hunk 6]
```diff
@@ -116,7 +116,9 @@ module.exports = {
             ['/general/architecture', 'Architecture'],
             ['/general/i18n', 'i18n'],
             ['/general/faceting', 'Faceting'],
-            ['/general/logging', 'Logging']
+            ['/general/logging', 'Logging'],
```
Comment (filrak) [inline packages/core/docs/.vuepress/config.js:119]:
```suggestion
            ['/general/logging', 'Logging']
```

**Review by filrak — COMMENTED** [review]
Missing docs for CT api client - please adjust the i18n part as well ;)

File: packages/core/docs/integrate/api-middleware.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  extendApi: (app) => {
```
Comment (filrak) [inline packages/core/docs/integrate/api-middleware.md:36]: maybe just `extend` as we have a whole express instance here therefore can do much more than just adding api endpoints ;)

Comment (filrak) [reply packages/core/docs/integrate/api-middleware.md:36]: or even better

**Pull request merged**